File: src/main/java/com/yogesh/meditrack/service/PatientService.java
package com.yogesh.meditrack.service;

import com.yogesh.meditrack.entity.Patient;

import java.util.List;

public interface PatientService {

    Patient savePatient(Patient patient);

    List<Patient> getAllPatients();

    Patient getPatientById(Long id);

    Patient updatePatient(Long id, Patient patient);

    void deletePatient(Long id);
}

File: frontend/src/service/patientService.js
import api from "./api";

export const getAllPatients = () => api.get("/admin/patients");

export const getPatientById = (id) => api.get(`/admin/patients/${id}`);

export const savePatient = (patient) => api.post("/admin/patients", patient);

export const updatePatient = (id, patient) =>
  api.put(`/admin/patients/${id}`, patient);

export const deletePatient = (id) => api.delete(`/admin/patients/${id}`);